shub/bootstrap.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import absolute_import
+
+import os
+import shutil
+import tempfile
+import zipfile
+
+import click
+import requests
+import yaml
+from click.formatting import HelpFormatter
+from six import BytesIO
+
+from shub.exceptions import (
+    BadParameterException, NotFoundException, RemoteErrorException)
+
+
+EXAMPLE_REPO = "scrapinghub/custom-images-examples"
+AVAILABLE_PROJECTS_URL = (
+    "https://raw.githubusercontent.com/%s/master/bootstrap_projects.yml"
+    "" % EXAMPLE_REPO)
+
+HELP = """
+Through custom images, Scrapinghub allows you to run crawlers written in any
+language you want. To get you started, we prepared a few examples projects in
+different programming languages and frameworks. You can find them in our custom
+images repository at:
+
+    https://github.com/scrapinghub/custom-images-examples
+
+The 'shub bootstrap' command clones an example project to the current directory
+so that you can start hacking right away.
+
+Run
+
+    shub bootstrap -l
+
+to get a list of all available example projects, then run
+
+    shub bootstrap PROJECTNAME
+
+to clone it.
+"""
+
+SHORT_HELP = "Clone custom image example project"
+
+
+def list_projects_callback(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return
+    projects = get_available_projects()
+    list_projects(projects)
+    ctx.exit()
+
+
+@click.command(help=HELP, short_help=SHORT_HELP)
+@click.option('-l', '--list', 'list_projects', help='list available projects',
+              is_flag=True, callback=list_projects_callback,
+              expose_value=False, is_eager=True)
+@click.argument('project')
+@click.argument('target_dir', required=False)
+def cli(project, target_dir):
+    target_dir = os.path.join(os.getcwd(), target_dir or project)
```

Comment:
I had thought about this and even implemented it at some point. In the end I decided for always cloning into a new subdir for two reasons:

* This is what `git clone` and `scrapy startproject` do
* It is less annoying to fix "it cloned into a subdir when I wanted the current dir" (by just moving everything one level up) than it is to fix "it cloned into the current dir but I wanted a subdir", because if you expect a subdir you will likely run the command in a generic folder where you keep multiple projects (e.g. I personally have `~/python/`), and now you have to figure out which files were there before and which were just cloned.

Although I am open to being convinced otherwise if you have a strong argument ;)

About the `normpath` I am curious, I've seen it in other projects too. Is there a case where `os.path.join(os.getcwd(), 'some_folder')` gives an invalid/unwanted path? From what I've gathered around the internets the "cleaning up redundant separators" part [can screw up paths that contain symlinks](https://stackoverflow.com/a/6394943).